### DISCUSSION TITLE
Red led stays on

### DISCUSSION THREAD

**Discussion by CrisesUrv** [darthcloud/BlueRetro#68] [Q&A]
I'm trying to setup an adapter exclusively for Dreamcast and two players. I've succesfully flashed the BlueRetro_dc_spiffs.bin and wired two DC extensions cables as detailed in the instructions. Since I'm only using P1 and P2 ports, I've wired IO18, IO23, IO26 & IO27 to 3.3V. Also, since this is a hardcoded firmware, I've *not wired* I39 to 3.3V and IO25, IO32, IO33, I34 & I35 to GND.

The thing is that the red led in the board always stays on, either when connected via usb to the PC (no connection to the console) or when connected to the console. In the serial debug, I get some errors and a loop o messages:

![Sin título1](https://user-images.githubusercontent.com/49307050/132962714-8bb94fe9-4e63-446f-a838-526b948e9cea.png)

Pushing the BOOT or EN buttons does not seem to produce any effect.

Any clues?

**Comment by darthcloud** [comment]
Logs look fines, which LED are you referring to? You added a LED on IO17 yourself? Base ESP32 only got a power LED which is always on.

**Reply by sudo-lupus666** [reply]
The red light pulses when the console is turned on and it blinks for a fraction of a second when "en" is pressed, doesn't it? I'm having a hard time myself with Dreamcast adaptor and it seems just like the case. I checked the 5v via and the console is delivering 4.75. I wonder if GDEmu has something to do with it because when I tried with the original GD drive, the adaptor worked flawlessly.

**Reply by darthcloud** [reply]
4.7 should be good in any case, If GDEMU got a menu its most likely that it's polling controller in a non standard way and BlueRetro dont like it. I don't have one so I can't verify/fixit.

**Reply by sudo-lupus666** [reply]
That's somehow strange. I've got it working after I started a game with the conventional gamepad and changed it to the Bluetooth adaptor after the video game console reboot (after the game was selected by GDEMU). Played a lot of MS6, a game converted from Naomi. I say that it's strange because I've got a PSIO and the PS1 adaptor works flawlessly. But, yes, got it working. Thank you!

**Reply by darthcloud** [reply]
I guess I need to get a GDemu now! Thanks for the update.

**Reply by sudo-lupus666** [reply]
One more update! I got the adaptor working with other games (not the ones converted from Naomi). I played Tomb Raider 5 for about 15 minutes and played a little bit of Alone in the Dark, too. No problems with them. Maybe the problem was the games I was testing? I returned to Naomi games (KOF XI this time) and I got an error/console turning off. Maybe the problem is the game and not the firmware. I will play more Dreamcast games through GDemu and I will give you more feedback. One more time, thank you very much for your great contribution to the scene.

**Reply by Downcraft** [reply]
Hello, I have a GDEMU too. I think it might be the "not standard games",  which are known to be buggy, as they are compromised ports.

I played loads of different games started by GDEMU without any problem using blueretro. 

I am not aware how a physical  diskdrive emulator should have any impact on controller behaviour. 

I can only imagine that some timing issues on controller input could lead to crashing naomi or other ported games. 

GDEMU itself should be not relevant I think, maybe I am wrong.

**Reply by darthcloud** [reply]
Well GDemu need some software to run its menu. All official games use a standard library for accessing the controllers.

I don't know anything about GDemu but its probably not using official library, and so the way it access the maple bus will differ slightly.

BlueRetro likely deviate from the standard quite a bit but officials software is fine with it. GDemu likewise likely deviate in some way but real Ctrl are fine with it. But plug both BlueRetro together and they are too far appart in timing to work.

Its probably not much to fix, but I need to get a GDemu first to look at it.

As for naomi game, issue for those is not related to GDemu, but its likely same type issue.

**Reply by Downcraft** [reply]
Ok I unverstand your point. Just wanting to say that controlling gdemu works flawless for me. I even tried different firmware versions. Maybe I am just lucky, for whatever reason.

**Reply by darthcloud** [reply]
Ok good to know!

I got a VA0, so I need to find another DC first before I can try.

**Comment by CrisesUrv** [comment]
Yes, I was referring to the IO17 led. I've resoldered it and now it behaves as expected: it pulses searching for a controller. 

However, I've tried to connect a DS4 (v1) with no luck: I force the DS4 pairwing with the PS+share button so that it blinks white, in a few seconds it gets a solid blue light but, then, in less than two seconds, it switches off. During the two seconds it seems to stay connected (with the blue light on), the controller is actually working, but the blueretro led stays pulsing all the time. I've repeated the pairing several times but it always behave the same.

**Reply by darthcloud** [reply]
Try with FW v0.19.1 rather than v1 beta.

**Reply by CrisesUrv** [reply]
The same thing happens, I'm affraid.

**Reply by darthcloud** [reply]
Can you try with another controller?

**Reply by CrisesUrv** [reply]
The only other bluetooth controller I have is a Steam controller (with BLE firmware), but that does not even connect. Both the Steam and DS4 controllers pair ok with a PC and the blueretro receiver also connects ok with the web interface, allowing to change its settings.

**Reply by darthcloud** [reply]
Can you take a picture of the DS4 sticker a the bottom so I can compare version number with my own?

**Reply by CrisesUrv** [reply]
Here it is:

![IMG_20210913_100206](https://user-images.githubusercontent.com/49307050/133046431-504492e3-8810-4a9a-a2b8-3360d9ae376a.jpg)

It's a v1 controller. So far I was able to pair it with a PC, a raspberry pi 3 and 4 and a 8bitdo bluetooth dongle.

**Reply by darthcloud** [reply]
Thanks, also once it paired once (turn blue) and then disconnect, can you connect back by simply pressing PS button?

If that fail, press Boot button once to exit pairing mode (led stop flash) try again to reconnect by only pressing PS button.

**Reply by CrisesUrv** [reply]
Tested all options with similar results: 
- Blueretro in pairing mode (red pulsating led) + DS4 into pairing mode (PS+sharing buttons, white led in DS4 flashing rapidly): pairs (blue led in DS4) for two seconds, then disconnects (no light in DS4).
- After that, DS4 into connect mode (PS button, white led flashnig slowly): pairs (blue led) for two seconds, then disconnects.
- Pressed Boot button on blueretro (red led stop pulsating) + DS4 into connect (PS button): pairs (blue led) for two seconds then disconnects. Immediately, blueretro atomatically starts pairing again (led pulsating).

Just to add: during the two seconds connection, the controller works in the dreamcast, but the red led in the blueretro adapter does not stop pulsating.

**Reply by darthcloud** [reply]
[v1.0-beta-wip1.zip](https://github.com/darthcloud/BlueRetro/files/7158114/v1.0-beta-wip1.zip)
Wild guess, give it a shot and let me know.

I any case I ordered a few V1 DS4 to try myself.

**Reply by CrisesUrv** [reply]
Same behaviour, I'm affraid. Thanks a lot for your efforts, anyway!

**Reply by CrisesUrv** [reply]
Hi!, just to update that the issue about connecting a DS4 V1 to a DC has been fixed in version 1.1! 

Everything seems to work well expect rumble: when enabled via the web interface, it gets detected by the games and works for some time (very weakly, nevertheless), but it disapears after a while and does not recover until the console (and the adapter) are reseted. Anyway I understand rumble support is experimental. Thanks!

**Reply by darthcloud** [reply]
I think commit 621c11f360758d83bc35020c6e3a549221791e39 is what fixed it. But not sure how you could get in a state for this to make an issue. I tested various V1 without reproducing.

I any case it's good that it's fixed.

Dreamcast rumble support is junk currently, It's poorly documented and I will need to do some proper RE on a real one one day to fix this. Better leave it off.
